### DISCUSSION TITLE
Reduce bundle size of `extension-drag-handle`

### DISCUSSION THREAD

**Discussion by RomainLanz** [ueberdosis/tiptap#5787] [Feature Requests]
### Description

Hey there! 👋🏻

I noticed that `tiptap-pro/extension-drag-handle` includes a direct dependency on `y-prosemirror`, which in turn depends on `yjs`. This setup significantly increases the bundle size, even if collaborative features aren’t being used.

Would it be possible to adjust the code for `@tiptap-pro/extension-drag-handle` to remove this direct dependency?

Unfortunately, I can’t review the code myself as the package isn’t open-source. Thanks for considering!

### Use Case

I want to avoid to bundle ~300Kb of extra JS.

### Type

Other

**Comment by nperez0111** [comment]
Unfortunately the extension does need y-prosemirror in order to work in collaborative scenarios. And I don't see a way around that.

300Kb sounds like a lot to me, are you sure about that? Some should definitely be tree-shakable.

**Reply by RomainLanz** [reply]
Yes, it is indeed quite large! https://pkg-size.dev/y-prosemirror

![image](https://github.com/user-attachments/assets/69f8d092-8919-4e68-ba94-812681ba1897)

(The ~300kb comes from Vite Bundle Analyzer, but it seems to take a bit more into account)

Would you consider refactoring the extension so that collaborative functionality is provided as a separate extension? This way, the core can stay lean, and collaboration features can be added as needed.
